### PR TITLE
Splitup methods in process_chain.py

### PR DESCRIPTION
**Pull request by anikaweinmann**
Part of https://github.com/mundialis/actinia_core/issues/189 which spilts up some methods in process_chain.py.

### CODE REVIEW

File: src/actinia_core/core/common/process_chain.py [hunk 1]
```diff
@@ -478,169 +520,232 @@ def _create_module_process(self, module_descr):
...
+                param = "%s=%s" % (param, value)
+            params.append(param)
+            # save the output dict in a resource export list
+            if "export" in output:
+                self.resource_export_list.append(output)
 
-        if ("interface-description" in module_descr
-                and module_descr["interface-description"] is True):
-            params.append("--interface-description")
+    def _check_stdout(self, module_descr, id):
```
Comment (griembauer) [inline src/actinia_core/core/common/process_chain.py:702]: Perspectively, maybe we could move all `_check...`, `_create...` etc. methods together?

**Review by griembauer — APPROVED** [review]
Thanks! Just one small remark for possible future re-structuring. Please merge

File: src/actinia_core/core/common/process_chain.py [hunk 2]
```diff
@@ -478,169 +520,232 @@ def _create_module_process(self, module_descr):
...
+        """Helper method to check the stdout in the module description and addes it
```
Comment (mmacata) [inline src/actinia_core/core/common/process_chain.py:703]: `add`

File: src/actinia_core/core/common/process_chain.py [hunk 3]
```diff
@@ -438,38 +506,12 @@ def _create_module_process(self, module_descr):
...
-                raise AsyncProcessError(
-                    "Missing *format* in stdout parser description of process id %s"
-                    % str(id))
-            if "delimiter" not in module_descr["stdout"]:
-                raise AsyncProcessError(
-                    "Missing *delimiter* in stdout parser description of process id %s"
-                    % str(id))
-
-            self.output_parser_list.append({id: module_descr["stdout"]})
+            self._check_stdout(module_descr, id)
```
Comment (mmacata) [inline src/actinia_core/core/common/process_chain.py:514]: Maybe rename to `_check_and_add_stdout` ? Or even `_check_and_get_stdout_command`  to make it similar to the other ones if you think it fits.

File: src/actinia_core/core/common/process_chain.py [hunk 4]
```diff
@@ -478,169 +520,232 @@ def _create_module_process(self, module_descr):
...
-                                   '<', '>', '\\']
+            return p
 
-                        for symbol in symbols:
-                            value = value.replace(symbol, " ")
+        return None
 
-                        values = value.split(" ")
+    def _add_grass_module_input_parameter_to_list(self, module_descr, params, id):
+        """Helper method to set the input parameter of a grass module and add
```
Comment (mmacata) [inline src/actinia_core/core/common/process_chain.py:567]: `parameters`

File: src/actinia_core/core/common/process_chain.py [hunk 5]
```diff
@@ -478,169 +520,232 @@ def _create_module_process(self, module_descr):
...
+                        for subentry in entry.split(","):
+                            if "@" in subentry:
+                                mapset = subentry.split("@")[1]
+                                if mapset not in self.required_mapsets:
+                                    self.required_mapsets.append(mapset)
+            params.append(param)
+
+    def _add_grass_module_output_parameter_to_list(self, module_descr, params, id,
+                                                   module_name):
+        """Helper method to set the output parameter of a grass module and add
```
Comment (mmacata) [inline src/actinia_core/core/common/process_chain.py:629]: `parameters`

**Review by mmacata — APPROVED** [review]
I mainly found some typos, nothing really preventing the code to run successful.
Really cool, thank you, I am curious how the code looks like now without seeing it as diff :)

**Pull request merged**